deep_qa/data/instances/mc_tuples_with_background_instance.py

```diff
@@ -0,0 +1,358 @@
+from collections import defaultdict
+from typing import Dict, List
+
+import numpy as np
+from overrides import overrides
+
+from .instance import TextInstance, IndexedInstance
+from ..data_indexer import DataIndexer
+
+
+class TextTuple:
+    def __init__(self, subject: str, verb: str, objects: List[str],
+                 context: str=None, time: str=None, loc: str=None, source: str=None):
+        self.subject = subject
+        self.verb = verb
+        self.objects = objects
+        self.context = context
+        self.time = time
+        self.location = loc
+        self.source = source
+
+    def __str__(self):
+        # NOTE: not currently displaying locaton, time, and source info
+        return "TextTuple( [\n\tsubj: {0} \n\tverb: {1} \n\tobjects: {2} \n\tcontext: {3}\n] )"
+
+    def to_text_list(self, object_handling: str="collapse", include_context: bool=True):
+        # Subject and verb
+        text_list = [self.subject, self.verb]
+        # Object(s)
+        if object_handling == "collapse":
+            text_list.append(" ".join(self.objects))
+        elif object_handling == "first":
+            if len(self.objects) > 0:
+                text_list.append(self.objects[0])
+        else:
+            raise Exception("Requested object_handling ({0}) not currently supported.".format(object_handling))
+        # Context
+        if include_context and self.context:
+            text_list.append(self.context)
+        # Currently, source, time and location aren't being used, but this can be changed in the future.
+
+        return text_list
+
+
+    @classmethod
+    def from_string(cls, input_string: str):
+        tuple_fields = input_string.lower().split("|")
+        num_fields = len(tuple_fields)
+        assert num_fields >= 2, "num_fields not >= 2" + input_string
+        subject = tuple_fields[0]
+        verb = tuple_fields[1]
+        objects = []
+        context, time, location, source = None, None, None, None
```

Comment:
changed, and I didn't know you could do this, but for the record this bothers me that python works like this, I'd think that that would make them all pointers to the same thing, s.t. if you change one you change them all.  Guess I still don't get pass-by-reference!